pkgs/development/compilers/arm-frc-linux-gnueabi-gcc/default.nix

```diff
@@ -0,0 +1,92 @@
+{stdenv, fetchurl, arm-frc-linux-gnueabi-binutils, arm-frc-linux-gnueabi-eglibc, arm-frc-linux-gnueabi-linux-api-headers, elfutils,
+libmpc, gmp, mpfr, zlib}:
+
+stdenv.mkDerivation rec {
+  _target = "arm-frc-linux-gnueabi";
+
+  version = "4.9.4";
+  name = "${_target}-gcc-${version}";
+
+  src = fetchurl {
+    url = "ftp://gcc.gnu.org/pub/gcc/releases/gcc-${version}/gcc-${version}.tar.bz2";
+    sha256 = "6c11d292cd01b294f9f84c9a59c230d80e9e4a47e5c6355f046bb36d4f358092";
+  };
+
+  patches =
+  [
+    ./minorSOname.patch
+    ./no-nested-deprecated-warnings.patch
+    ./format-error-fix.patch
```

Comment:
An alternative would have been to just disable format warnings `hardeningDisable = [ "format" ];`